custom/icds/views/ccz_hosting.py

```diff
@@ -0,0 +1,265 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.http import HttpResponseRedirect
+from django.utils.translation import (
+    ugettext_lazy,
+    ugettext_noop,
+    ugettext as _,
+)
+from django.utils.functional import cached_property
+from django.utils.decorators import method_decorator
+from django.urls import reverse
+from django.shortcuts import redirect
+from django.views.generic import TemplateView
+from django.contrib import messages
+from django.http import HttpResponse
+from django.conf import settings
+
+from couchexport.models import Format
+from corehq import toggles
+from corehq.apps.domain.auth import get_username_and_password_from_request
+from corehq.apps.app_manager.dbaccessors import (
+    get_brief_apps_in_domain,
+    get_build_by_version,
+)
+from corehq.apps.domain.views import (
+    BaseDomainView,
+    DomainViewMixin,
+)
+from corehq.apps.hqwebapp.decorators import use_select2_v4
+from corehq.apps.locations.permissions import location_safe
+from corehq.apps.domain.decorators import login_and_domain_required
+from corehq.util.download import get_download_response
+from custom.icds.forms import (
+    CCZHostingForm,
+    CCZHostingLinkForm,
+)
+from custom.icds.models import (
+    CCZHostingLink,
+    CCZHosting,
+)
+from custom.icds.utils.ccz_hosting import CCZHostingUtility
+
+
+@location_safe
+@method_decorator([login_and_domain_required, toggles.MANAGE_CCZ_HOSTING.required_decorator()], name='dispatch')
+class ManageCCZHostingLink(BaseDomainView):
+    urlname = "manage_ccz_hosting_links"
+    page_title = ugettext_lazy("Manage CCZ Hosting Links")
+    template_name = 'icds/manage_ccz_hosting_links.html'
+    section_name = ugettext_lazy("CCZ Hosting Links")
+
+    @cached_property
+    def section_url(self):
+        return reverse(ManageCCZHostingLink.urlname, args=[self.domain])
+
+    @cached_property
+    def form(self):
+        return CCZHostingLinkForm(
+            domain=self.domain,
+            data=self.request.POST if self.request.method == "POST" else None
+        )
+
+    def get_context_data(self, **kwargs):
+        links = [l.to_json() for l in CCZHostingLink.objects.filter(domain=self.domain)]
+        return {
+            'form': self.form,
+            'links': links,
+            'domain': self.domain,
+        }
+
+    def delete(self):
+        link_id = self.kwargs.get('link_id')
+        if link_id:
+            try:
+                ccz_hosting_link = CCZHostingLink.objects.get(pk=link_id)
+            except CCZHostingLink.DoesNotExist:
+                pass
+            else:
+                ccz_hosting_link.delete()
```

Comment:
Shouldn't this line go in the `try`?

Reply:
You mean a new try block?
fwiw, i dont expect `delete` to fail.

Reply:
Ignore this, I was confused about exactly what `else` does. Your code is good.